packages/core/src/exchanges/cache.ts

```diff
@@ -86,6 +86,7 @@ export const cacheExchange: Exchange = ({ forward, client }) => {
         ),
       ]),
       map(op => addMetadata(op, { cacheOutcome: 'miss' })),
+      filter(op => !(op.operationName === 'query' && op.context.requestPolicy === 'cache-only')),
```

Comment:
Woah, this was a bug in the default cache too? 😱

Reply:
```suggestion
      filter(op => op.operationName !== 'query' || op.context.requestPolicy !== 'cache-only'),
```